lib/v2/nju/jw.js

```diff
@@ -1,45 +1,57 @@
 const got = require('@/utils/got');
-const cheerio = require('cheerio');
 const { parseDate } = require('@/utils/parse-date');
 const timezone = require('@/utils/timezone');
 
 module.exports = async (ctx) => {
     const type = ctx.params.type;
     const type_dict = {
-        ggtz: ['https://jw.nju.edu.cn/ggtz/list.htm', '公告通知'],
-        jxdt: ['https://jw.nju.edu.cn/_t1143/24774/list.psp', '教学动态'],
+        ggtz: [
+            26263, // columnID
+            '公告通知', // type name
+            '[{"field":"top","type":"desc"},{"field":"new","type":"desc"},{"field":"publishTime","type":"desc"}]', // order
+            '[{"field":"title","name":"title"},{"field":"f1","name":"f1"},{"field":"publishTime","pattern":[{"name":"d","value":"yyyy-MM-dd HH:mm:ss"}],"name":"publishTime"},{"field":"link","name":"link"}]', // returnInfos
+        ],
+        jxdt: [
+            24774,
+            '教学动态',
+            '[{"field":"top","type":"desc"},{"field":"new","type":"desc"},{"field":"publishTime","type":"desc"}]',
+            '[{"field":"title","name":"title"},{"field":"publishTime","pattern":[{"name":"d","value":"yyyy-MM-dd HH:mm:ss"}],"name":"publishTime"},{"field":"link","name":"link"}]',
+        ],
     };
-    const response = await got({
-        method: 'get',
-        url: type_dict[type][0],
+    const { data } = await got({
+        method: 'post',
+        url: 'https://jw.nju.edu.cn/_wp3services/generalQuery?queryObj=articles',
+        form: {
+            siteId: 414,
+            columnId: type_dict[type][0],
+            pageIndex: 1,
+            rows: 24, // 用大一点的值，因为前面太多置顶的
+            orders: type_dict[type][2],
+            returnInfos: type_dict[type][3],
+        },
+        headers: {
+            Origin: 'https://jw.nju.edu.cn',
+            Referer: 'https://jw.nju.edu.cn/main.htm',
+        },
     });
 
-    const data = response.data;
-
-    const $ = cheerio.load(data);
-    const list = $('li.news');
-
     ctx.state.data = {
         title: `本科生院-${type_dict[type][1]}`,
-        link: type_dict[type][0],
-        item: list
-            .map((index, item) => {
-                item = $(item);
+        link: `https://jw.nju.edu.cn/${type_dict[type][0]}/list.htm`,
```

Comment:
https://jw.nju.edu.cn/26263/list.htm will not work, it should be https://jw.nju.edu.cn/ggtz/list.htm